src/image_function.cpp

```diff
@@ -747,6 +747,30 @@ namespace Image_Function
         }
     }
 
+    std::vector < uint32_t > Histogram( const Image & image, const Image & mask )
+    {
+        const Image maskedImage = BitwiseAnd( image, mask );
```

Comment:
The idea behind mask is to do not calculate some areas of image, meaning to do not include results from these areas of image. If we use bitwise logical AND the number of results in histogram remains the same. What we need is to add pixel values from original image only on areas where mask has non 0 values. For example, we found a circle on an image and we want a histogram only this circle so we use mask for this.

Reply:
The code in new function should be something like this:
```cpp
        const uint32_t rowSize = image.rowSize();
        const uint32_t rowSizeMask = mask.rowSize();

        const uint8_t * imageY     = image.data() + y * rowSize + x;
        const uint8_t * imageYMask = mask.data() + yMask * rowSizeMask + xMask;
        const uint8_t * imageYEnd  = imageY + height * rowSize;

        for( ; imageY != imageYEnd; imageY += rowSize, imageYMask += rowSizeMask ) {
            const uint8_t * imageX     = imageY;
            const uint8_t * imageXMask = imageYMask;
            const uint8_t * imageXEnd  = imageX + width;

            for( ; imageX != imageXEnd; ++imageX, ++imageXMask ) {
                if( (*imageXMask ) > 0 )
                    ++histogram[*imageX];
            }
        }
```